### PR TITLE
plugins/aerial: init

### LINKED ISSUE #2514: [PLUGIN REQUEST] aerial.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | `aerial.nvim`  | 
| Homepage | https://github.com/stevearc/aerial.nvim  | 
| Nixpkgs  | [vimPlugins.aerial-nvim](https://github.com/nix-community/nixvim/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=%5BPLUGIN+REQUEST%5D+%3CPLUGIN%3E) | 

## Extra Information

It presents a better code outline

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [aerial.nvim](https://github.com/stevearc/aerial.nvim), a plugin for a code outline window.

Fixes #2514

### CODE REVIEW

File: plugins/by-name/aerial/default.nix [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+  name = "aerial";
+  originalName = "aerial.nvim";
+  package = "aerial-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsOptions =
+    let
+      size = with types; either ints.unsigned (numbers.between 0.0 1.0);
+      listOfSize = with types; either size (listOf size);
```
Comment (MattSturgeon) [inline plugins/by-name/aerial/default.nix:16]: Is this always supposed to be a 2-elem list? Should we be using our custom `listOfLen` type?

File: plugins/by-name/aerial/default.nix [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+
+      # Several options can have a single value, or a filetype-value mapping.
+      mkTypeOrAttrsOfType =
+        type: default: description:
+        defaultNullOpts.mkNullable (with types; either type (attrsOf type)) default description;
```
Comment (MattSturgeon) [inline plugins/by-name/aerial/default.nix:21]: This has no "raw" lua support, either at the top-level or at any elem-types.

File: plugins/by-name/aerial/default.nix (hunk 1 above)
Comment (MattSturgeon) [reply plugins/by-name/aerial/default.nix:16]: Also, this has no raw-lua support

**Review by MattSturgeon — COMMENTED** [review]
No major issues.

Minor issues:
- most of the custom types have no raw-lua support
- do we _really_ need this many settings options? :sweat:

**Comment by khaneliman** [comment]
Yeah, if we're going to document this many options it'd be nice to have rawLua support for everything, in case we don't map things up properly.

**Review by MattSturgeon — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Pull request merged**